sdk/java-sdk-spring/src/it/java/com/example/wiring/valueentities/user/UserEntity.java

```diff
@@ -49,7 +50,8 @@ public Effect<User> getUser() {
 
   @PostMapping("/{email}/{name}")
   public Effect<String> createOrUpdateUser(@PathVariable String email, @PathVariable String name) {
-    return effects().updateState(new User(email, name)).thenReply("Ok");
+    return effects().updateState(new User(email, name)).thenReply("Ok",
+        Metadata.EMPTY.withHttpResponseCode(StatusCode.Success.CREATED));
```

Comment:
pretty long name:) maybe `.withResponseCode` and the fact that it's HTTP can be derived from the parameter, also, similar like here:
```
      <T> Effect<T> error(String description, Status.Code grpcErrorCode);
      <T> Effect<T> error(String description, StatusCode.ErrorCode httpErrorCode);
```

Reply:
ok, yeah.. now I'm in doubt between `.withStatusCode` or `.withResponseCode`.. wikipedia calls them [HTTP status codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes). so, maybe `.withStatusCode`?

Reply:
yeah `withStatusCode`